Python/flowgraph.c

```
@@ -1406,33 +1411,67 @@ nop_out(basicblock *bb, int start, int count)
     }
 }
 
+/* Determine opcode & oparg for freshly folded constant.
+   Steals reference to "newconst".
+*/
+static int
+newop_from_folded(PyObject *newconst, PyObject *consts,
+                  PyObject *const_cache, int *newopcode, int *newoparg)
+{
+    if (PyLong_CheckExact(newconst)) {
+        int overflow;
+        long val = PyLong_AsLongAndOverflow(newconst, &overflow);
+        if (!overflow && _PY_IS_SMALL_INT(val)) {
+            *newopcode = LOAD_SMALL_INT;
+            *newoparg = val;
+            return SUCCESS;
+        }
+    }
+    *newopcode = LOAD_CONST;
+    *newoparg = add_const(newconst, consts, const_cache);
+    RETURN_IF_ERROR(*newoparg);
+    return SUCCESS;
+}
+
+/* Steals reference to "newconst" */
+static int
+make_const(PyObject *newconst, basicblock *bb,
+           int i, int nop, PyObject *consts, PyObject *const_cache)
+{
+    if (newconst == NULL) {
+        if (PyErr_ExceptionMatches(PyExc_KeyboardInterrupt)) {
+            return ERROR;
+        }
+        PyErr_Clear();
```

We don't have context about the exception here, so this is not the place to swallow it. The null check should happen in the caller.

---

Do you suggest `eval_const_unaryop` & `eval_const_binop` should have this logic?

---

Generally, the return value of a function should be checked for error immediately after the function call.